src/Microsoft.AspNetCore.Mvc.TagHelpers/ValidationMessageTagHelper.cs

```diff
@@ -89,7 +91,7 @@ public ValidationMessageTagHelper(IHtmlGenerator generator)
                         // We check for whitespace to detect scenarios such as:
                         // <span validation-for="Name">
                         // </span>
-                        var childContent = await output.GetChildContentAsync();
+
                         if (childContent.IsEmptyOrWhiteSpace)
                         {
                             // Provide default message text (if any) since there was nothing useful in the Razor source.
```

Comment:
Should have mentioned:
- This comment is no longer correct since message may have come from the Razor source now
- The `else` block starting at [line 103](https://github.com/aspnet/Mvc/pull/8087/files#diff-44deef455d0ca749c788544788e1b252R103) is no longer needed
- The comment above about whitespace belongs above the `GenerateValidationMessage(...)` call

I suspect this block could now be narrowed to:
``` c#
if (!output.IsContentModified && tagBuilder.HasInnerHtml))
{
    output.Content.SetHtmlContent(tagBuilder.InnerHtml);
}
```